node/core/pvf/execute-worker/src/lib.rs

```diff
@@ -148,9 +185,11 @@ pub fn worker_entrypoint(
 					artifact_path.display(),
 				);
 
+				if !artifact_path.starts_with(cache_path) {
+					return Err(io::Error::new(io::ErrorKind::Other, format!("received an artifact path {artifact_path:?} that does not belong to expected artifact dir {cache_path:?}")))
+				}
```

Comment:
TODO: I wonder if an io::Error is the most appropriate error here. To change it we would have to change the error type of this function (would need a new error kind) which would be a bit annoying. But since this would be a logic bug from our side it can be an `assert!` instead.